service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/database/DumpAzureDatabaseStep.java

```diff
@@ -0,0 +1,134 @@
+  private final LandingZoneApiDispatch landingZoneApiDispatch;
+  private final SamService samService;
+  private final WorkspaceService workspaceService;
+  private final AzureStorageAccessService azureStorageAccessService;
+  private final AzureDatabaseUtilsRunner azureDatabaseUtilsRunner;
+
+  public DumpAzureDatabaseStep(
+      ControlledAzureDatabaseResource sourceDatabase,
+      LandingZoneApiDispatch landingZoneApiDispatch,
+      SamService samService,
+      WorkspaceService workspaceService,
+      AzureStorageAccessService azureStorageAccessService,
+      AzureDatabaseUtilsRunner azureDatabaseUtilsRunner) {
+    this.sourceDatabase = sourceDatabase;
+    this.landingZoneApiDispatch = landingZoneApiDispatch;
+    this.samService = samService;
+    this.workspaceService = workspaceService;
+    this.azureStorageAccessService = azureStorageAccessService;
+    this.azureDatabaseUtilsRunner = azureDatabaseUtilsRunner;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    var inputParameters = context.getInputParameters();
+    var workingMap = context.getWorkingMap();
+    var destinationWorkspaceId =
+        inputParameters.get(
+            WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_WORKSPACE_ID, UUID.class);
+    var userRequest =
+        getRequired(
+            inputParameters,
+            JobMapKeys.AUTH_USER_INFO.getKeyName(),
+            AuthenticatedUserRequest.class);
+
+    var destinationContainer =
+        getRequired(
+            workingMap,
+            WorkspaceFlightMapKeys.ControlledResourceKeys.AZURE_STORAGE_CONTAINER,
+            ControlledAzureStorageContainerResource.class);
+
+    workingMap.put(
+        WorkspaceFlightMapKeys.ControlledResourceKeys.AZURE_STORAGE_CONTAINER,
+        destinationContainer);
+
+    var sasToken =
+        azureStorageAccessService.createAzureStorageContainerSasToken(
+            destinationWorkspaceId, destinationContainer, userRequest, null, null, "rcw");
+    var blobContainerUrlAuthenticated = sasToken.sasUrl();
+
+    var blobFileName =
+        String.format(
+            "%s_%s.dump",
+            sourceDatabase.getDatabaseName(),
+            new SimpleDateFormat("yyyy-MM-dd_HH:mm:ss").format(new java.util.Date()));
+    workingMap.put(WorkspaceFlightMapKeys.ControlledResourceKeys.CLONE_DB_DUMPFILE, blobFileName);
+
+    // Query LZ for the postgres server info
+    var bearerToken = new BearerToken(samService.getWsmServiceAccountToken());
+    var landingZoneId =
+        landingZoneApiDispatch.getLandingZoneId(
+            bearerToken, workspaceService.getWorkspace(sourceDatabase.getWorkspaceId()));
+    var dbServerName =
+        getResourceName(
+            landingZoneApiDispatch
+                .getSharedDatabase(bearerToken, landingZoneId)
+                .orElseThrow(() -> new RuntimeException("No shared database found")));
```

Comment:
sonarcloud wants us to replace these with a "dedicated exception" - what do you think?

Reply:
yes, WSM team says: create a new type (or use IllegalStateException)